### PR TITLE
INT-4853 - Add new properties to `snyk_project`

### PR DESCRIPTION
**Pull request by austinkelleher**
Improved Snyk project name parser to allow for new properties to
be added on `snyk_project`

- New properties added to entities:

  | Entity         | Properties         |
  | -------------- | ------------------ |
  | `snyk_project` | `repoFullName`     |
  | `snyk_project` | `repoOrganization` |
  | `snyk_project` | `repoName`         |
  | `snyk_project` | `directoryName`    |
  | `snyk_project` | `fileName`         |

### CODE REVIEW

File: src/steps/projects/codeRepo.ts [hunk 1]
```diff
@@ -9,16 +9,93 @@ function isSupportedCodeRepoOrigin(projectOrigin: string): boolean {
...
+  // `projectName` can be in two different formats:
+  // Input 1. `starbase-test/starbase:subdir/package.json`
+  // Input 2. `starbase-test/starbase:package.json`
+  //
+  // Output 1. ['starbase-test/starbase', 'subdir/package.json']
+  // Output 2. ['starbase-test/starbase', 'package.json']
+  const [repoFullName, fileScannedPath] = projectName.split(':');
+  if (!repoFullName) return getUnknownSnykProjectNameParseResult();
+
+  // `starbase-test/starbase` => `starbase`
```
Comment (VDubber) [inline src/steps/projects/codeRepo.ts:78]: This format is common for github repo full names. What about bitbucket or other vcs tools?

File: src/steps/projects/codeRepo.ts [hunk 2]
```diff
@@ -9,16 +9,93 @@ function isSupportedCodeRepoOrigin(projectOrigin: string): boolean {
...
+  if (fileScannedPath) {
+    ({ directoryName, fileName } = parseSnykProjectFileScannedPath(
+      fileScannedPath,
+    ));
+  }
+
+  return {
+    repoFullName: repoFullName.toLowerCase(),
+    repoOrganization: repoOrganization.toLowerCase(),
+    repoName: repoName.toLocaleLowerCase(),
```
Comment (VDubber) [inline src/steps/projects/codeRepo.ts:95]: is locale intentional here?

File: src/steps/projects/codeRepo.ts [hunk 3]
```diff
@@ -9,16 +9,93 @@ function isSupportedCodeRepoOrigin(projectOrigin: string): boolean {
...
+ * {
+ *   repoFullName: 'starbase-test/starbase',
+ *   repoOrganization: 'starbase-test',
+ *   repoName: 'starbase',
+ *   directoryName: 'subdir',
+ *   fileName: 'package.json'
+ * }
+ */
+function parseSnykProjectName(projectName: string): ParseSnykProjectNameResult {
+  // `projectName` can be in two different formats:
```
Comment (VDubber) [inline src/steps/projects/codeRepo.ts:69]: The comments/docs are really appreciated

**Review by VDubber — DISMISSED** [review]
Love the code cleanup. Left a few minor comments.

File: src/steps/projects/codeRepo.ts (hunk 2 above)
Comment (austinkelleher) [reply src/steps/projects/codeRepo.ts:95]: Good catch. Fixing.

File: src/steps/projects/codeRepo.ts (hunk 1 above)
Comment (austinkelleher) [reply src/steps/projects/codeRepo.ts:78]: Good question. I'm not entirely sure and it doesn't seem to be documented. I do think we should check the format of others separately.

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**